packages/web/pages/stake/index.tsx

```diff
@@ -71,33 +85,94 @@ export const Staking: React.FC = observer(() => {
     return { currency: osmo, amount: primitiveAmount.amount, denom: osmo };
   }, [osmo, primitiveAmount]);
 
+  const delegationQuery = cosmosQueries.queryDelegations.getQueryBech32Address(
+    account?.address ?? ""
+  );
+
+  const userValidatorDelegations = delegationQuery.delegations;
+
+  const usersValidatorsMap = useMemo(() => {
+    const delegationsMap = new Map<string, StakingType.Delegation>();
+
+    userValidatorDelegations.forEach((delegation: StakingType.Delegation) => {
+      delegationsMap.set(delegation.delegation.validator_address, delegation);
+    });
+
+    return delegationsMap;
+  }, [userValidatorDelegations]);
+
+  const [amountDefault, setAmountDefault] = useState<AmountDefault>("input");
+
+  const getProperties = useCallback(() => {
+    const amountUSD = priceStore
+      .calculatePrice(
+        new CoinPretty(
+          osmo,
+          new Dec(amountConfig.amount).mul(
+            DecUtils.getTenExponentNInPrecisionRange(osmo.coinDecimals)
+          )
+        )
+      )
+      ?.toString();
+
+    const amount = amountConfig.amount;
+
+    const properties: Properties = {
+      amountDefault,
+      amount,
+      amountUSD,
+    };
+
+    return properties;
+  }, [amountConfig.amount, amountDefault, osmo, priceStore]);
+
   const stakeCall = useCallback(() => {
+    const properties = getProperties();
+
+    logEvent([EventName.Stake.stakingStarted, properties]);
+
     if (account?.address && account?.osmosis && coin?.amount) {
-      account.osmosis.sendDelegateToValidatorSetMsg(coin);
+      account.osmosis.sendDelegateToValidatorSetMsg(
+        coin,
+        "",
+        (tx: any) =>
+          Boolean(tx?.code) &&
```

Comment:
Nit but will add more type safety:

The base `signAndBroadcast` used in `sendDelegateToValidatorSetMsg` returns a `DeliverTxResponse` type for the `tx` passed to onFullfill.

In our osmosis account functions (here `sendDelegateToValidatorSetMsg`), we could increase type safety by using that type instead of washing to any; like `(tx: any) => ...`. Both in the osmosis account message and here.

Future devs may also benefit from using the type to discover the structure of the tx, like the fact that it has an error code.